### PR TITLE
[dong][BE] Check transaction status for TransferMoney #847

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
#847

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

**Review by vietdangquoc — APPROVED** [review]

File: dong-service/services/p2p_order_service.go [hunk 1]
```diff
@@ -315,32 +315,45 @@ func (s *OrderService) ConfirmOrderAsSeller(ctx context.Context, orderID int64,
 
 		if intermediaryWallet != nil && o.OrderAmount.Sign() > 0 {
 			txHash, transferErr := s.blockchain.TransferMoney(intermediaryWallet.EncryptedPrivateKey, *offer.IntermediaryWalletAddress, *targetWallet, o.OrderAmount.String(), fmt.Sprintf("%s Order %d", constants.TextDataP2PTrading, orderID), extraInfo)
+			
+			transferTxHash = &txHash
+
 			if transferErr != nil {
+				if o.OfferID != nil {
```
Comment (hoangdohuy) [inline dong-service/services/p2p_order_service.go:322]: khả năng là bị duplicate với logic nhánh điều kiện xử lý fail 
=> Giữ 1 trong 2 thôi

**Review by vietdangquoc — APPROVED** [review]

**Review by hoangdohuy — APPROVED** [review]
LGTM
